pkg/api/file.go

```diff
@@ -262,19 +262,30 @@ func (s *server) fileDownloadHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	outBuffer := bytes.NewBuffer(nil)
-	c, err := file.JoinReadAll(j, e.Reference(), outBuffer)
-	if err != nil && c == 0 {
+	pr, pw := io.Pipe()
+
+	go func() {
+		_, err := file.JoinReadAll(j, e.Reference(), pw)
```

Comment:
what about a wait or a channel to close in this goroutine to signal that it exited?

Reply:
That could be done by closing the reader, I suppose? On defer or context done?

Reply:
@acud @pradovic I have improved on this cancelation.